metis-core/metis-core-service/src/test/java/eu/europeana/metis/core/dao/TestDepublishRecordIdDao.java

```diff
@@ -0,0 +1,178 @@
+        .find(DepublishRecordId.class).delete(new DeleteOptions().multi(true));
+    reset(provider, depublishRecordIdDao);
+  }
+
+  @BeforeAll
+  static void prepare() {
+    embeddedLocalhostMongo = new EmbeddedLocalhostMongo();
+    embeddedLocalhostMongo.start();
+    String mongoHost = embeddedLocalhostMongo.getMongoHost();
+    int mongoPort = embeddedLocalhostMongo.getMongoPort();
+    mongoClient = MongoClients
+        .create(String.format("mongodb://%s:%s", mongoHost, mongoPort));
+    provider = spy(new MorphiaDatastoreProviderImpl(mongoClient, "test"));
+    depublishRecordIdDao = spy(
+        new DepublishRecordIdDao(provider, MAX_DEPUBLISH_RECORD_IDS_PER_DATASET, PAGE_SIZE));
+  }
+
+  @AfterAll
+  static void destroy() {
+    embeddedLocalhostMongo.stop();
+  }
+
+  @Test
+  void createRecordIdsToBeDepublishedTest() throws BadContentException {
+    final String datasetId = Integer.toString(TestObjectFactory.DATASETID);
+    final Set<String> setTest = new HashSet<>();
+    setTest.add("1001");
+
+    depublishRecordIdDao.createRecordIdsToBeDepublished(datasetId, setTest);
+
+    assertEquals(1, provider.getDatastore().find(DepublishRecordId.class).count());
+  }
+
+  @Test
+  void deletePendingRecordIdsTest() throws BadContentException {
+    final String datasetId = Integer.toString(TestObjectFactory.DATASETID);
+    final Set<String> setTest = new HashSet<>();
+    setTest.add("1002");
+    final Query<DepublishRecordId> mockQuery = mock(Query.class);
+    final Datastore mockDatastore = mock(Datastore.class);
+
+    depublishRecordIdDao
+        .addRecords(setTest, datasetId, DepublicationStatus.DEPUBLISHED, Instant.now());
+
+    doReturn(mockDatastore).when(provider).getDatastore();
+    doReturn(mockQuery).when(mockDatastore).find(DepublishRecordId.class);
+    doReturn(1L).when(depublishRecordIdDao).deleteRecords(mockQuery);
+    depublishRecordIdDao.deletePendingRecordIds(datasetId, setTest);
+
+    verify(mockDatastore, times(1)).find(DepublishRecordId.class);
+    verify(mockQuery, times(3)).filter(any());
+    assertEquals(0, provider.getDatastore().find(DepublishRecordId.class).count());
+  }
+
+
+  @Test
+  void countSuccessfullyDepublishedRecordIdsForDatasetTest() {
+    final String datasetId = Integer.toString(TestObjectFactory.DATASETID);
+    final Set<String> setTest = new HashSet<>();
+    setTest.add("1003");
+
+    depublishRecordIdDao
+        .addRecords(setTest, datasetId, DepublicationStatus.DEPUBLISHED, Instant.now());
+    reset(provider);
+    long result = depublishRecordIdDao.countSuccessfullyDepublishedRecordIdsForDataset(datasetId);
+
+    verify(provider, times(1)).getDatastore();
+    assertEquals(1L, result);
+  }
+
+  @Test
+  void getDepublishRecordIdsTest() {
+    final String datasetId = Integer.toString(TestObjectFactory.DATASETID);
+    final Set<String> setTest = new HashSet<>();
+    setTest.add("1004");
+    setTest.add("1005");
```

Comment:
You can also do `final Set<String> setTest = Set.of("1004", "1005");`
The same goes I guess for the other methods.

Reply:
Indeed that is much simpler, will do that